coordinator/coordinator.go

```diff
@@ -276,24 +270,32 @@ func (c *Coordinator) WatchForNewJobs() error {
 	c.Logger.Info("Watching for new jobs")
 	// TODO: change it so we can watch for new jobs using task manager; searching for PENDING status tasks
 	// JOB__TYPE__NAME__VARIANT
-	runs, err := c.taskManager.GetAllTaskRuns()
+	runs, err := c.Metadata.Tasks.GetAllRuns()
 	if err != nil {
 		return fferr.NewInternalError(err)
 	}
-	runs.FilterByStatus(scheduling.Pending)
+
+	runs.FilterByStatus(scheduling.PENDING)
 
 	for _, run := range runs {
 		go func(run scheduling.TaskRunMetadata) {
-			// LOCK and pass the key to the worker; if lock fails then skip
-			// UNLOCK after the worker is done
-			// LockTaskRun(taskID TaskID, runId TaskRunID) (sp.LockObject, error)
-			lock, err := c.taskManager.LockTaskRun(run.TaskId, run.ID)
+			lock, err := c.Locker.LockRun(run.ID)
```

Comment:
nit: just thinking, I wonder if it makes sense for the LockRun to return the unlock function that one just has to defer

Reply:
thats a good idea